content/blog/2021-02-17-disabled-features.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Introducing Disabled Features
+created_at: 2021-02-17
+kind: article
+author_name: Ganesh Vernekar
+---
+
+We have always made hard promises around stability and breaking changes following the SemVer model. That will remain to be the case.
+
+As we want to be bolder in experimentation, we are planning to use feature flags more.
+
+Starting v2.25.0, we have introduced a new section called [disabled features](https://prometheus.io/docs/prometheus/latest/disabled_features/) which have the features hidden behind the `--enable-feature` flag. You can expect more and more features getting added to this section in the future releases.
+
+The features in this list are considered experimental and comes with following considerations as long as they are still behind `--enable-feature`:
+
+1. API specs can change if the feature has any API (web API, code interfaces, etc.).
+2. The behaviour of the feature can change.
+3. They can break some assumption that you might have had about Prometheus.
+    * For example the assumption that a query does not look ahead of the evaluation time for samples, which will be broken by `@` modifier and negative offset.
+4. They are allowed to be unstable while we try to keep them as stable as possible.
+
+These considerations allow us to be more bold with experimentation and innovate. Once any feature gets widely used and is considered stable with respect to its API, behaviour, and implementation, they may be enabled by default and removed from disabled features list. If we find any feature to be not worth it, we many completely remove it. If some feature is considered a big breaking change for Prometheus, it might stay disabled until maybe a v3.0.0 release.
```

Comment:
we many completely remove it --> we _may_ completely remove it.


until maybe a v3.0.0 release --> until a v3.0.0 release